### PR TITLE
Display interface cleanup

### PR DESCRIPTION
**Pull request by cepetr**
This pull request is about a small cleanup of the display interface:

**This PR does not bring any functional changes**

1. Removes the unused display_utf8_substr() function
2. Renames display_text_width(), display_text_split() and moves them to font.h
3. Introduces a new terminal.h/.c where terminal printing function is moved.
4. Introduces a new display_utils.c/.h where some functions are moved.
5. Renames lib/display_interface.h to trezorhal/display.h.
6. ~~Temporarily renames lib/display.h to lib/display_draw.h.~~
~~Number 6 is preparation for further display interface refactoring.~~

### CODE REVIEW

File: core/SConscript.bootloader_ci [hunk 1]
```diff
@@ -74,7 +74,9 @@ CPPPATH_MOD += [
         'vendor/micropython/lib/uzlib',
 ]
 SOURCE_MOD += [
-    'embed/lib/display.c',
+    'embed/lib/display_draw.c',
+    'embed/lib/display_utils.c',
+    'embed/lib/terminal.c',
```
Comment (TychoVrahe) [inline core/SConscript.bootloader_ci:79]: lets sort this like the other Sconscripts

File: core/embed/trezorhal/display.h [hunk 2]
```diff
@@ -17,11 +17,12 @@
...
  */
 
-#ifndef _DISPLAY_INTERFACE_H
-#define _DISPLAY_INTERFACE_H
+#ifndef TREZORHAL_DISPLAY_H
+#define TREZORHAL_DISPLAY_H
 
 #include <stdint.h>
 #include "common.h"
+#include "display_draw.h"
```
Comment (TychoVrahe) [inline core/embed/trezorhal/display.h:25]: i know we talked about this before, but looking at it closer now i dislike this already, lets remove this from here and add it where necessary, its just few places and shouldn't take much effort.

**Review by TychoVrahe — COMMENTED** [review]
Looks great, two minor comments below.

The `rust/src/trezorhal/display.rs` could be splat into display.rs, font.rs and display_draw.rs to better match the underlying implementation. Not sure its worth doing in this PR.

File: core/SConscript.bootloader_ci (hunk 1 above)
Comment (cepetr) [reply core/SConscript.bootloader_ci]: I've sorted this section in all sconscripts, see 9cb7cb23918e51137acd82b42aa1f5e39db36b2d

**Review by TychoVrahe — APPROVED** [review]

**Pull request merged**